src/main/java/edu/wisc/my/personalizedredirection/controller/PersonalizedRedirectionController.java

```diff
@@ -77,14 +77,15 @@ public void setRedirectionService(IRedirectionService redirectionService) {
             if (url == null || url.length() != 0) {
                 response.sendRedirect(url);
             } else {
+                logger.error("No url found for app " + appName);
                 response.setStatus(
-                        HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+                        HttpServletResponse.SC_NOT_FOUND);
             }
 
         } catch (Exception e) {
             logger.error("Issues happened while trying to generate custom link",
                     e);
-            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
```

Comment:
Because INTERNAL_SERVER_ERROR threw you to an ugly page if the link wasn't found. With a 404, hopefully we're picked up our normal MyUW 404 page.